clarity/contracts/arkadiko-oracle-v2-4.clar

```diff
@@ -0,0 +1,252 @@
+  )
+)
+
+;; ---------------------------------------------------------
+;; Custom
+;; ---------------------------------------------------------
+
+(define-public (get-price-custom (key (string-ascii 32)))
+  (if (is-eq key "stSTX/USD")
+    (get-price-ststx)
+    (err ERR-PRICE-NOT-FOUND)
+  )
+)
+
+(define-public (get-price-ststx)
+  (let (
+    (stx-price (try! (get-price-pyth 0xec7a775f46379b5e943c3526b1c8d54cd49749176b0b98e02dde68d1bd335c17)))
+    (ststx-reserve 'SP4SZE494VC2YC5JYG7AYFQ44F5Q4PYV7DVMDPBG.reserve-v1)
+    ;; TODO: Update for mainnet
+    ;; (ratio (try! (contract-call? 'SP4SZE494VC2YC5JYG7AYFQ44F5Q4PYV7DVMDPBG.data-core-v2 get-stx-per-ststx ststx-reserve)))
+    (ratio (try! (contract-call? .data-core-v2 get-stx-per-ststx ststx-reserve)))
+  )
+    (ok (/ (* stx-price (* ratio u100)) u100000000))
+  )
+)
+
+;; ---------------------------------------------------------
+;; Manual
+;; ---------------------------------------------------------
+
+(define-public (update-price-manual (token (string-ascii 12)) (decimals uint) (price uint))
+  (begin
+    (asserts! (is-eq tx-sender (var-get manual-manager)) (err ERR-NOT-AUTHORIZED))
+    (map-set prices-manual { token: token } { decimals: decimals, price: price })
+    (ok price)
+  )
+)
+
+(define-public (set-manual-manager (new-manager principal))
+  (begin
+    (asserts! (is-eq tx-sender (var-get manual-manager)) (err ERR-NOT-AUTHORIZED))
+    (var-set manual-manager new-manager)
+    (ok new-manager)
+  )
+)
+
+
+;; ---------------------------------------------------------
+;; Init
+;; ---------------------------------------------------------
+
+;; Oracle prices are standardized to 8 decimal places (100000000).
+;; For tokens with 6 decimals (like STX), we divide by 100000000 to normalize.
+;; For tokens with 8 decimals (like xBTC/sBTC), we divide by an additional 100 in the Arkadiko protocol to account for the extra precision.
+
+(define-constant decimals8 u100000000)
+
+(begin
+  (map-set prices-pyth { token: "STX" } { decimals: decimals8, feed-id: 0xec7a775f46379b5e943c3526b1c8d54cd49749176b0b98e02dde68d1bd335c17 })
+  (map-set prices-pyth { token: "xSTX" } { decimals: decimals8, feed-id: 0xec7a775f46379b5e943c3526b1c8d54cd49749176b0b98e02dde68d1bd335c17 })
+
+  (map-set prices-pyth { token: "BTC" } { decimals: (* decimals8 u100), feed-id: 0xe62df6c8b4a85fe1a67db44dc12de5db330f7ac66b72dc658afedf0f4a415b43 })
+  (map-set prices-pyth { token: "xBTC" } { decimals: (* decimals8 u100), feed-id: 0xe62df6c8b4a85fe1a67db44dc12de5db330f7ac66b72dc658afedf0f4a415b43 })
+  (map-set prices-pyth { token: "sBTC" } { decimals: (* decimals8 u100), feed-id: 0xe62df6c8b4a85fe1a67db44dc12de5db330f7ac66b72dc658afedf0f4a415b43 })
+
+  (map-set prices-custom { token: "stSTX" } { decimals: decimals8, key: "stSTX/USD" })
+  (map-set prices-custom { token: "ststx-token" } { decimals: decimals8, key: "stSTX/USD" })
+
+  (map-set prices-dia { token: "DIKO" } { decimals: decimals8, key: "DIKO/USD" })
```

Comment:
Q: how does DIA set the price of DIKO?

It's probably not important since only Zest uses it as collateral (and you cannot borrow against it afaik), but just wondering... they probably use our DEX?

Reply:
A: from what I found they use BitFlow (which aggregates all dexes on stacks). 
https://www.diadata.org/blog/post/stacks-bitcoin-l2-dia-oracles/